source/reference/minio-server/settings/console.rst

```diff
@@ -176,6 +178,212 @@ Log Query URL
+      :sync: config
+
+      .. mc-conf:: browser csp_policy
+         :delimiter: " "
+
+         .. code-block:: shell
+            :class: copyable
+		 
+            mc admin config set browser \
+               csp_policy="default-src 'self' 'unsafe-eval' 'unsafe-inline';" \
+               [ARGUMENT=VALUE ...]
+
+
+Strict Transport Security
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+*Optional*
+
+Configure MinIO console to generate a `Strict-Transport-Security <https://en.wikipedia.org/wiki/HTTP_Strict_Transport_Security>`__ header in HTTP responses.
+
+To generate the header, you **must** set a duration using either :envvar:`MINIO_BROWSER_HSTS_SECONDS` or :mc-conf:`~browser.hsts_seconds`.
+Other HSTS settings are optional.
+
+.. tab-set::
+
+   .. tab-item:: Environment Variables
+      :sync: envvar
+
+      .. envvar:: MINIO_BROWSER_HSTS_INCLUDE_SUB_DOMAINS
+
+         Set to ``on`` to also apply the configured HSTS policy to all MinIO Console subdomains.
+         Defaults to ``off``.
+
+         .. code-block:: shell
+            :class: copyable
+
+            set MINIO_BROWSER_HSTS_INCLUDE_SUB_DOMAINS="on"
+
+      .. envvar:: MINIO_BROWSER_HSTS_PRELOAD
+
+         Set to ``on`` to direct the client browser to add the MinIO Console domain to its HSTS preload list.
+         Defaults to ``off``.
+
+         .. code-block:: shell
+            :class: copyable
+
+            set MINIO_BROWSER_HSTS_INCLUDE_SUB_DOMAINS="on"
+
+      .. envvar:: MINIO_BROWSER_HSTS_SECONDS
+
+         The ``max_age`` the configured policy remains in effect, in seconds.
+         Defaults to ``0``, disabled.
+         You **must** configure a *non-zero* duration to enable the ``Strict-Transport-Security`` header.
+
+         .. code-block:: shell
+            :class: copyable
+
+            set MINIO_BROWSER_HSTS_SECONDS=31536000
+
+   .. tab-item:: Configuration Settings
+      :sync: config
+
+      The following configuration settings require a service restart to take effect.
+      You can do this with :mc-cmd:`mc admin service restart`.
+
+      .. mc-conf:: browser hsts_include_subdomains
+         :delimiter: " "
+
+         Set to ``on`` to also apply the configured HSTS policy to all MinIO Console subdomains.
+         Defaults to ``off``.
+
+         .. code-block:: shell
+            :class: copyable
+
+            mc admin config set browser \
+               hsts_include_subdomains="on" \
+               hsts_seconds="31536000" \
+               [ARGUMENT=VALUE ...]
+
+      .. mc-conf:: browser hsts_preload
+         :delimiter: " "
+
+         Set to ``on`` to direct the client browser to add the MinIO Console domain to its HSTS preload list.
+         Defaults to ``off``.
+
+         .. code-block:: shell
+            :class: copyable
+
+            mc admin config set browser \
+               hsts_preload="on" \
+               hsts_seconds="31536000" \
+               [ARGUMENT=VALUE ...]
+
+      .. mc-conf:: browser hsts_seconds
```

Comment:
Consider putting this first in the list, since it's the one required to generate the header.